### PR TITLE
feat(odyssey-react): withStyles Title

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: packages/odyssey-react/src/components/Title/Title.tsx [hunk 1]
```diff
@@ -10,28 +10,23 @@
...
+import type { ComponentPropsWithoutRef, ReactText } from "react";
+import { forwardRef } from "react";
+import { useCx, useOmit, withStyles } from "../../utils";
 import styles from "./Title.module.scss";
 
-type Levels = "1" | "2" | "3" | "4" | "5" | "6";
-
-export interface Props
+interface Props
   extends Omit<ComponentPropsWithoutRef<"h1">, "style" | "className"> {
```
Comment (andrewberg-okta) [inline packages/odyssey-react/src/components/Title/Title.tsx:19]: Switch to `ComponentPropsWithRef` now that we have forward ref?

File: packages/odyssey-react/src/components/Title/Title.test.tsx [hunk 2]
```diff
@@ -40,12 +40,17 @@ describe("Title", () => {
...
-    const component = screen.getByRole(heading);
-    expect(component).toHaveAttribute("href");
+    // @ts-expect-error heading element does not have href attribute
+    <Title href="/foo/bar.baz" children={titleText} />;
+  });
+
+  it("invokes ref with expected args after render", () => {
+    const ref = jest.fn();
+
+    render(<Title children={titleText} />);
```
Comment (andrewberg-okta) [inline packages/odyssey-react/src/components/Title/Title.test.tsx:50]: Title needs the `ref` prop assigned for this test to pass

File: packages/odyssey-react/src/components/Title/Title.stories.tsx [hunk 3]
```diff
@@ -24,9 +24,11 @@ export default {
   },
 };
 
-const Template: Story<Props> = ({ level, visualLevel, children }) => (
-  <Title level={level} visualLevel={visualLevel} children={children} />
-);
+const Template: Story<ComponentProps<typeof Title>> = ({
```
Comment (andrewberg-okta) [inline packages/odyssey-react/src/components/Title/Title.stories.tsx:27]: Are we really moving to not publicly exposing props for components? `ComponentProps` appeared to be an escape hatch for cases where developers hadn't made a public api. Forcing downstream developers to get at our types this way feels cumbersome / like an antipattern.

File: packages/odyssey-react/src/components/Title/Title.tsx (hunk 1 above)
Comment (jeffbelser-okta) [reply packages/odyssey-react/src/components/Title/Title.tsx:19]: @andrewberg-okta the other way around in fact. Even without a ref favor `ComponentPropsWithoutRef` in all cases unless there is some other critical extending circumstance!

Comment (andrewberg-okta) [reply packages/odyssey-react/src/components/Title/Title.tsx:19]: Let me rephrase, we don't include them in the interface?
